packages/keyboard-shortcuts/src/context.js

```diff
@@ -3,4 +3,26 @@
  */
 import { createContext } from '@wordpress/element';
 
-export const context = createContext();
+const globalShortcuts = new Set();
+const globalListener = ( event ) => {
+	for ( const keyboardShortcut of globalShortcuts ) {
+		keyboardShortcut( event );
+	}
+};
+
+export const context = createContext( {
+	current: {
+		add: ( shortcut ) => {
+			if ( globalShortcuts.size === 0 ) {
+				document.addEventListener( 'keydown', globalListener );
+			}
+			globalShortcuts.add( shortcut );
```

Comment:
Very nice, more elegant than the code snippet I proposed. I like how the `Set` is also used as a reference counter.